### PR TITLE
[SPARK-24987][SS] - Fix Kafka consumer leak when no new offsets for TopicPartition

### PR DESCRIPTION
**Pull request by YuvalItzchakov**
## What changes were proposed in this pull request?

This small fix adds a `consumer.release()` call to `KafkaSourceRDD` in the case where we've retrieved offsets from Kafka, but the `fromOffset` is equal to the `lastOffset`, meaning there is no new data to read for a particular topic partition. Up until now, we'd just return an empty iterator without closing the consumer which would cause a FD leak.

If accepted, this pull request should be merged into master as well.

## How was this patch tested?

Haven't ran any specific tests, would love help on how to test methods running inside `RDD.compute`.

### CODE REVIEW

**Comment by YuvalItzchakov** [comment]
This is the same as https://github.com/apache/spark/pull/21983 only merged against master (after @felixcheung comment). Should be merged to branch-2.3.

**Comment by felixcheung** [comment]
ok to test

**Comment by felixcheung** [comment]
@koeninger back

File: external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceRDD.scala [hunk 1]
```diff
@@ -124,8 +124,6 @@ private[kafka010] class KafkaSourceRDD(
       thePart: Partition,
       context: TaskContext): Iterator[ConsumerRecord[Array[Byte], Array[Byte]]] = {
     val sourcePartition = thePart.asInstanceOf[KafkaSourceRDDPartition]
-    val topic = sourcePartition.offsetRange.topic
-    val kafkaPartition = sourcePartition.offsetRange.partition
```
Comment (felixcheung) [inline external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceRDD.scala]: I see these are unused...

**Review by felixcheung — APPROVED** [review]
this makes sense to me.
LGTM.

File: external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceRDD.scala (hunk 1 above)
Comment (YuvalItzchakov) [reply external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceRDD.scala]: Yes, removed them since they were unused.

**Review by koeninger — APPROVED** [review]
LGTM, merging to master and branch-2.3

**Pull request closed without merging**